source/slang-capture-replay/slang-session.cpp

```diff
@@ -159,38 +179,102 @@ namespace SlangCapture
+
+        TypeConformanceCapture* conformanceCapture = new TypeConformanceCapture(pConformance);
+        Slang::ComPtr<TypeConformanceCapture> resultCapture(conformanceCapture);
+        *outConformance = resultCapture.detach();
+
         return result;
     }
 
     SLANG_NO_THROW SlangResult SessionCapture::createCompileRequest(
         SlangCompileRequest**   outCompileRequest)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         SlangResult result = m_actualSession->createCompileRequest(outCompileRequest);
         return result;
     }
 
     SLANG_NO_THROW SlangInt SessionCapture::getLoadedModuleCount()
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         SlangInt count = m_actualSession->getLoadedModuleCount();
         return count;
     }
 
     SLANG_NO_THROW slang::IModule* SessionCapture::getLoadedModule(SlangInt index)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         slang::IModule* pModule = m_actualSession->getLoadedModule(index);
+
+        if (pModule)
+        {
+            ModuleCapture* moduleCapture = m_mapModuleToCapture.tryGetValue(pModule);
+            if (!moduleCapture)
+            {
+                assert(!"Module not found in mapModuleToCapture");
+            }
+            return static_cast<slang::IModule*>(moduleCapture);
+        }
+
         return pModule;
     }
 
     SLANG_NO_THROW bool SessionCapture::isBinaryModuleUpToDate(const char* modulePath, slang::IBlob* binaryModuleBlob)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         bool result = m_actualSession->isBinaryModuleUpToDate(modulePath, binaryModuleBlob);
         return result;
     }
 
+    ModuleCapture* SessionCapture::getModuleCapture(slang::IModule* module)
+    {
+        ModuleCapture* moduleCapture = nullptr;
+        moduleCapture = m_mapModuleToCapture.tryGetValue(module);
+        if (!moduleCapture)
+        {
+            moduleCapture = new ModuleCapture(module);
+            Slang::ComPtr<ModuleCapture> result(moduleCapture);
+            m_mapModuleToCapture.add(module, *result.detach());
+        }
+        return moduleCapture;
+    }
+
+    SlangResult SessionCapture::getActualComponentTypes(
+            slang::IComponentType* const*   componentTypes,
+            SlangInt                        componentTypeCount,
+            List<slang::IComponentType*>&   outActualComponentTypes)
+    {
+        for (SlangInt i = 0; i < componentTypeCount; i++)
+        {
+            slang::IComponentType* const& componentType = componentTypes[i];
```

Comment:
The assertion thing I pointed out above can cause a problem here in release build because componentTypes will have zero size.

Reply:
fix that assert issue.